main.tf

```diff
@@ -35,7 +35,7 @@ resource "aws_apprunner_service" "this" {
   }
 
   dynamic "instance_configuration" {
-    for_each = length(var.instance_configuration) > 0 ? [var.instance_configuration] : []
+    for_each = length(var.instance_configuration) > 0 ? [var.instance_configuration] : (var.create_instance_iam_role ? [{ instance_role_arn = aws_iam_role.instance[0].arn }] : [])
```

Comment:
this shouldn't need to change

Reply:
I can reverse this change bu want to offer a concrete example why this change was made. Scenario:
1. You want to create an apprunner service with secrets
1. You want to leave the instance config's CPU and memory to defaults.

Therefore, you set `create_instance_iam_role` to `true`, understanding that it is needed in order to work with secrets.

With the current code, in order to get get this to work, you will now need to do the following in the relevant app service module:

```hcl
create_instance_iam_role = true
instance_configuration = {
    instance_role_arn = "garbage value"
}
```
i.e. you **must** provide a non-empty instance configuration, but as you want to leave CPU/mem to defaults, instead a garbage value needs to be provided to `instance_role_arn` - and the current code will overwrite this with the correct ARN.

I think this is pretty counter-intuitive . With the suggested change, instead you can simply do:

```hcl
create_instance_iam_role = true
```
...and the code creates the `instance_configuration` object with its usual defaults unless you provided one yourself of course.

Arguably of course one should always specify CPU/memory for production usage, but on the other hand that's not enforced as an input across the module anywhere else as far as I know. Anyway lmk your thoughts

Reply:
ah ok, thank you for that detailed explanation. Looks like there was some flaws in the initial default logic so I have corrected that here so you don't need to provide a "dummy" value just to get Terraform to pick up the changes